### PR TITLE
Test-pad Improvements

### PR DESCRIPTION
**Pull request by dereknola**
Changes the follow around test-pad tool:
- Correct spelling errors and improve documentation around using test-pad
- Convert flag parsing to gnu `getopt`, allows long and short flags
- Add support for debian11 as an OS, nice to have a nft native iptables OS

### CODE REVIEW

File: test-pad/test-pad [hunk 1]
```diff
@@ -162,61 +163,68 @@ get_possible_node_roles() {
...
+    *) echo "Unexpected flag: $1"
+        usage; exit 1;;
     esac
 done
 
 has_vagrant
 has_curl
 has_wget
 has_awk
+has_getopt
```
Comment (brooksn) [inline test-pad/test-pad:226]: Does this need to appear above PARSED_ARGUMENTS?

**Review by brooksn — APPROVED** [review]

File: test-pad/test-pad [hunk 2]
```diff
@@ -253,25 +261,28 @@ elif  [ "${REPO}" = "k3s" ]; then
...
+    OS=generic/ubuntu2004
 else 
    echo "error: unknown repo (-r) $REPO"
    exit 1
 fi
 
-if [ "${OS}" = "alpine" ]; then 
-    OS=generic/alpine316; 
-elif [ "${OS}" = "ubuntu" ]; then 
+if [ "${OSFLAG}" = "alpine" ]; then
```
Comment (briandowns) [inline test-pad/test-pad:277]: If this still contains the flag, the variable name of OS is more descriptive unless this contains a flag and then the rename would be accurate.

Comment (dereknola) [reply test-pad/test-pad:277]: I am trying to make it easier on the user so they don't have to type the full OS name and allows OS upgrades behind-the-scenes. 

So OSFLAG is the flag passed by the user: `alpine` `ubuntu` `debian`.
The OS is the actual true name of the vagrant box: `generic/alpine316`, `generic/ubuntu2004`, `generic/debian11`

Do you have a suggestion on better naming?

File: test-pad/test-pad (hunk 1 above)
Comment (dereknola) [reply test-pad/test-pad:226]: Good catch, I have move the `has_XXX` up to occur first

File: test-pad/test-pad (hunk 2 above)
Comment (briandowns) [reply test-pad/test-pad:277]: Nah, that works. I must have missed that nuance. :)

**Review by briandowns — APPROVED** [review]

**Pull request merged**